extensions/wikia/Search/classes/Result/ResultHelper.php

```diff
@@ -1,67 +1,81 @@
+use Wikia\Search\MediaWikiService, Wikia\Search\Utilities, CommunityDataService, ImagesService, PromoImage;
 
 class ResultHelper
 {
-	const MAX_WORD_COUNT_EXACT_MATCH = 16;
+	const MAX_WORD_COUNT_EXACT_MATCH = 40;
 	const MAX_WORD_COUNT_XWIKI_RESULT = 60;
 
 	/**
-	 * +     * Extends search result with additional data from outside search index, like description and image
-	 * +     *
-	 * +     * @param $result
-	 * +     * @param $pos
-	 * +     * @param $descWordLimit
-	 * +     * @return array
-	 * +     */
-	public static function extendResult($result, $pos, $descWordLimit)
-	{
-		$commData = new CommunityDataService($result['id']);
+	 * Extends search result with additional data from outside search index, like description and image
+	 *
+	 * @param $result
+	 * @param $pos
+	 * @param $descWordLimit
+	 * @param $imageSizes
+	 * @param query
+	 * @return array
+	 */
+	public static function extendResult( $result, $pos, $descWordLimit, $imageSizes, $query = null ) {
+		$commData = new CommunityDataService( $result['id'] );
 		$imageURL = ImagesService::getImageSrc(
 			$result['id'],
 			$commData->getCommunityImageId(),
-			WikiaSearchController::CROSS_WIKI_PROMO_THUMBNAIL_WIDTH,
-			WikiaSearchController::CROSS_WIKI_PROMO_THUMBNAIL_HEIGHT)['src'];
+			$imageSizes['width'],
+			$imageSizes['height']
+		)['src'];
 
 		$thumbTracking = "thumb";
 		//Fallback: if Curated Mainpage is inaccessible, try to use Special:Promote
 		//TODO: Remove after DAT-3642 is done
-		if (empty($imageURL)) {
-			$imageFileName = PromoImage::fromPathname($result['image_s'])->ensureCityIdIsSet($result['id'])->getPathname();
+		if ( empty( $imageURL ) ) {
+			$imageFileName = PromoImage::fromPathname( $result['image_s'] )
+				->ensureCityIdIsSet( $result['id'] )
+				->getPathname();
 			$imageURL = ImagesService::getImageSrcByTitle(
-				(new \CityVisualization)->getTargetWikiId($result['lang_s']),
+				( new \CityVisualization)->getTargetWikiId( $result['lang_s'] ),
 				$imageFileName,
-				WikiaSearchController::CROSS_WIKI_PROMO_THUMBNAIL_WIDTH,
-				WikiaSearchController::CROSS_WIKI_PROMO_THUMBNAIL_HEIGHT);
+				$imageSizes['width'],
+				$imageSizes['height']
+			);
 		}//TODO: end
 
-		if (empty($imageURL)) {
+		if ( empty( $imageURL ) ) {
 			// display placeholder image if no thumbnail
 			$imageURL = \F::app()->wg->ExtensionsPath . '/wikia/Search/images/wiki_image_placeholder.png';
 			$thumbTracking = "no-thumb";
 		}
 
-		$description = $result->limitTextLength($commData->getCommunityDescription(), $descWordLimit);
-		$description = !empty($description)
+		$description = $result->limitTextLength( $commData->getCommunityDescription(), $descWordLimit );
+		$description = !empty( $description )
 			? $description
-			: $result->getText(Utilities::field('description'), $descWordLimit);
+			: $result->getText( Utilities::field( 'description' ), $descWordLimit );
 
 		$service = new MediaWikiService();
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused "$service" local variable. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1481)